File: radar-commons-server/src/main/java/org/radarbase/stream/collector/RecordCollector.java
package org.radarbase.stream.collector;

import org.apache.avro.generic.IndexedRecord;

public interface RecordCollector {
    /**
     * Add a sample to the collection.
     * @param record new sample that has to be analysed
     * @throws IllegalStateException if this collector was constructed without a schema.
     */
    RecordCollector add(IndexedRecord record);
}

File: radar-commons-server/src/main/java/org/radarbase/stream/collector/RecordCollector.kt
package org.radarbase.stream.collector

import org.apache.avro.generic.IndexedRecord

interface RecordCollector {
    /**
     * Add a sample to the collection.
     * @param record new sample that has to be analysed
     * @throws IllegalStateException if this collector was constructed without a schema.
     */
    fun add(record: IndexedRecord): RecordCollector
}
